### PR TITLE
Tests to illustrate corner cases in cmd_args parsing

### PR DESCRIPTION
**Pull request by ringerc**
Use test cases to illustrate the quirks and corner cases of spiffe-helper's csv-based argument vector parser, per discussion in https://github.com/spiffe/spiffe-helper/pull/243 and to extract the argument vector case from the more complex test case in my slush-pile at https://github.com/ringerc/spiffe-helper-PRs/blob/31a134b2438ea865739673779c23c99beb410815/pkg/sidecar/sidecar_test.go#L116-L131

### CODE REVIEW

File: pkg/sidecar/sidecar_test.go [hunk 1]
```diff
@@ -252,6 +252,61 @@ func TestGetCmdArgs(t *testing.T) {
...
+			expectedArgs: []string{"echo", "Mc'Gougall"},
+		},
+		// Unlike a shell, spiffe-helper will parse this argument
+		// string without considering the single quoted range as a
+		// single argument.
+		{
+			name:         "single quotes do not protect spaces",
+			in:           `-c 'echo hello world'`,
+			expectedArgs: []string{"-c", "'echo", "hello", "world'"},
+		},
```
Comment (ringerc) [inline pkg/sidecar/sidecar_test.go:273]: Test case to illustrate https://github.com/spiffe/spiffe-helper/pull/243#pullrequestreview-2591877988

**Comment by ringerc** [comment]
Any chance of merging this pretty straight forward test case?

**Review by faisal-memon — APPROVED** [review]
Thanks @ringerc for making these additional tests.
